notification-portlet-webapp/src/main/webapp/scripts/modal-notice.js

```diff
@@ -0,0 +1,162 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * This file leverages Modal Bootstrap to display a single notice, like an Term of Service
+ */
+
+var upmodal_notice = upmodal_notice || {};
+
+if (!upmodal_notice.init) {
+
+  upmodal_notice.init = true;
+
+  (function() {
+
+    var defaults = {
+      selectors: {
+        content:         '.np-content',
+        title:           '.np-title',
+        form:            '.np-action-form',
+        body:            '.np-body',
+        link:            '.np-link',
+        actions:         '.np-actions',
+        actionTemplate:  '.np-action-template',
+        closeButton:     '.np-close'
+      }
+    };
+
+    // First 'prime-the-pump' with an ActionURL
+    var initNotices = function($, settings, callback) {
+      $.ajax({
+        type:    'POST',
+        url:     settings.invokeNotificationServiceUrl,
+        async:   false,
+        success: function() {
+          fetchNotices($, settings, callback);
+        }
+      });
+    }
+
+    // Then fetch the notifications with a ResourceURL
+    var fetchNotices = function($, settings, callback) {
+      $.ajax({
+        url:      settings.getNotificationsUrl,
+        type:     'POST',
+        dataType: 'json',
+        success:  function (data) {
+          feed = data.feed;
+          callback(feed);
+        },
+        error:    function () {
+          container.html(" ").text("AJAX failed. ~ THE END ~");
+        }
+      });
+    }
+
+    upmodal_notice.launch = function ($, container, options) {
+
+      var settings = $.extend({}, defaults, options);
+
+      var handleAction = function() {
+        var actionButton = $(this);
+        var formElement = container.find(settings.selectors.form);
+        $.post(actionButton.attr('data-url'), formElement.serialize(), function(data, textStatus, jqXHR) {
+          if (console && console.log) {
+            console.log('handleAction received status: ' + textStatus);
+          }
+        });
+        // Any action closes the modal window
+        container.modal('hide');
+      }
+
+      var resetDialog = function() {
+        container.off('hidden.bs.modal'); // Clear previous event handler(s)
+        container.find(settings.selectors.closeButton).hide();
+        container.find(settings.selectors.actions).find('.np-action').remove();
+      }
+
+      var drawActions = function(alert) {
+        var actionsContainer = container.find(settings.selectors.actions);
+        var availableActions = alert.availableActions;
+        var actionTemplate = actionsContainer.find(settings.selectors.actionTemplate);
+        if (availableActions && availableActions.length > 0) {
+          // Draw actions if we have them...
+          for (var i=0; i < availableActions.length; i++) {
```

Comment:
:art: space before and after assignment `=` operator.

Reply:
Writing it this way inside a `for` loop initialization is idiomatic.